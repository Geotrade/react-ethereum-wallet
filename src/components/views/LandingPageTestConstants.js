export const providerConstants = {
  Geth: {
    disabled: false,
    image: 'mist-150x150.png',
    link: 'https://github.com/ethereum/go-ethereum/releases',
  },
  Parity: {
    disabled: false,
    image: 'parity-signer.svg',
    link: 'https://github.com/paritytech/parity-ethereum/releases',
  },
  Metamask: {
    disabled: false,
    image: 'metamask-icon.svg',
    link: 'https://metamask.io/',
  },
  Infura: {
    disabled: false,
    image: 'infura-icon.jpeg',
    link: 'https://infura.io/',
  },
  Ganache: {
    disabled: false,
    image: 'ganache-icon.png',
    link: 'https://truffleframework.com/docs/ganache/quickstart',
  },
};
