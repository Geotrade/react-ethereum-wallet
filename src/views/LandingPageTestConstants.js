export const providerConstants = {
  Geth: {
    disabled: false,
    image: 'mist-150x150.png',
    link: 'https://github.com/ethereum/go-ethereum/releases',
  },
  Parity: {
    disabled: false,
    image: 'parity-signer.svg',
    link: 'https://github.com/paritytech/parity-ethereum/releases',
  },
  Metamask: {
    disabled: false,
    image: 'metamask-icon.svg',
    link: 'https://metamask.io/',
  },
  Infura: {
    disabled: false,
    image: 'infura-icon.jpeg',
    link: 'https://infura.io/',
  },
  Ganache: {
    disabled: false,
    image: 'ganache-icon.png',
    link: 'https://truffleframework.com/docs/ganache/quickstart',
  },
};

export const networks = {
  MainNet: {
    disabled: false,
    type: 'PoW',
  },
  Rinkeby: {
    set provider(prov) {
      this.selectedProvider = prov;
    },
    get disabled() {
      return this.selectedProvider === 'Parity';
    },
    type: 'PoA / Clique',
  },
  Ropsten: {
    disabled: false,
    type: 'PoW',
  },
  Kovan: {
    set provider(prov) {
      this.selectedProvider = prov;
    },
    get disabled() {
      return this.selectedProvider === 'Geth';
    },
    type: 'PoA / Clique',
  },
  Sokol: {
    set provider(prov) {
      this.selectedProvider = prov;
    },
    get disabled() {
      return this.selectedProvider !== 'Parity';
    },
    type: 'PoA / Clique',
  },
  GÃ¶rli: {
    set provider(prov) {
      this.selectedProvider = prov;
    },
    get disabled() {
      return this.selectedProvider !== 'Parity';
    },
    type: 'PoA / Clique',
  },
  INFURAnet: {
    set provider(prov) {
      this.selectedProvider = prov;
    },
    get disabled() {
      return this.selectedProvider !== 'Infura';
    },
    type: 'PoA / Clique',
  },
};
