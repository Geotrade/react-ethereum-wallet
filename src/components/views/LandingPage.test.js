import React from 'react';
import renderer from 'react-test-renderer';
import * as Constants from './LandingPageTestConstants.js';

describe('landing page links', () => {
  Object.keys(Constants.providerConstants).map(prov => {
    it('renders correctly`${prov}`', () => {
      const tree = renderer
        .create(
          <li>
            <a
              href={Constants.providerConstants[prov].link}
              target="_blank"
              rel="noopener noreferrer"
            >
              {prov}
            </a>
          </li>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
