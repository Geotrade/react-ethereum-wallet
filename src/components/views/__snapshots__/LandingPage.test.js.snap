// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing page links renders correctly\`\${prov}\` 1`] = `
<li>
  <a
    href="https://github.com/ethereum/go-ethereum/releases"
    rel="noopener noreferrer"
    target="_blank"
  >
    Geth
  </a>
</li>
`;

exports[`landing page links renders correctly\`\${prov}\` 2`] = `
<li>
  <a
    href="https://github.com/paritytech/parity-ethereum/releases"
    rel="noopener noreferrer"
    target="_blank"
  >
    Parity
  </a>
</li>
`;

exports[`landing page links renders correctly\`\${prov}\` 3`] = `
<li>
  <a
    href="https://metamask.io/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Metamask
  </a>
</li>
`;

exports[`landing page links renders correctly\`\${prov}\` 4`] = `
<li>
  <a
    href="https://infura.io/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Infura
  </a>
</li>
`;

exports[`landing page links renders correctly\`\${prov}\` 5`] = `
<li>
  <a
    href="https://truffleframework.com/docs/ganache/quickstart"
    rel="noopener noreferrer"
    target="_blank"
  >
    Ganache
  </a>
</li>
`;
